### PR TITLE
Allow building the generator image locally instead of pulling existing one

### PR DESCRIPTION
**Pull request by Kukks**

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # This script will run docker-compose-generator in a container to generate the yml files
 
-docker pull btcpayserver/docker-compose-generator
+If (-not ($BTCPAYGEN_DOCKER_IMAGE)) { $BTCPAYGEN_DOCKER_IMAGE = "btcpayserver/docker-compose-generator" }
+If($BTCPAYGEN_DOCKER_IMAGE -eq "build"){
+	docker build docker-compose-generator --tag build
```
Comment (NicolasDorier) [inline build.ps1]: use `btcpayserver/docker-compose-generator:local` instead of `build`.

File: build.ps1 [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+} Else {
+	docker pull $BTCPAYGEN_DOCKER_IMAGE:
```
Comment (NicolasDorier) [inline build.ps1]: why the `:`?

File: build.sh [hunk 3]
```diff
@@ -1,7 +1,13 @@
 #!/bin/bash
 
+if [ "$BTCPAYGEN_DOCKER_IMAGE" == "build" ]
+then
+    docker build docker-compose-generator --tag build
```
Comment (NicolasDorier) [inline build.sh]: use btcpayserver/docker-compose-generator:local instead of build.

File: build.ps1 [hunk 4]
```diff
@@ -1,6 +1,12 @@
 # This script will run docker-compose-generator in a container to generate the yml files
 
-docker pull btcpayserver/docker-compose-generator
+If (-not ($BTCPAYGEN_DOCKER_IMAGE)) { $BTCPAYGEN_DOCKER_IMAGE = "btcpayserver/docker-compose-generator" }
+If($BTCPAYGEN_DOCKER_IMAGE -eq "build"){
```
Comment (NicolasDorier) [inline build.ps1]: use `btcpayserver/docker-compose-generator:local` instead of `build`.

File: build.sh [hunk 5]
```diff
@@ -1,7 +1,13 @@
 #!/bin/bash
 
+if [ "$BTCPAYGEN_DOCKER_IMAGE" == "build" ]
```
Comment (NicolasDorier) [inline build.sh]: use `btcpayserver/docker-compose-generator:local` instead of `build`

**Comment by NicolasDorier** [comment]
Super cool I wanted that. Some suggestion to make the code simpler.

File: build.ps1 (hunk 2 above)
Comment (Kukks) [reply build.ps1]: brainfart

**Comment by Kukks** [comment]
@NicolasDorier 
Regarding the review:
I updated the builder segment as follows:
```
if [ "$BTCPAYGEN_DOCKER_IMAGE" == "btcpayserver/docker-compose-generator:local" ]
then
    docker build docker-compose-generator --tag $BTCPAYGEN_DOCKER_IMAGE
else
    docker pull ${BTCPAYGEN_DOCKER_IMAGE:-"btcpayserver/docker-compose-generator"}
fi
```

Is that enough or had more changes in mind to it?

**Comment by Kukks** [comment]
done @NicolasDorier deet

**Pull request merged**